### ISSUE TITLE
Increase minimal CA Key size for RSA keys

### ISSUE THREAD

**Issue by XolphinMartijn** [cabforum/smime#242]
Our current requirements sets the key size for CA certificates at 2048 bits for RSA. Given how long Root and Sub CA certificates may live, I would suggest we add language to raise this to a minimum of 3072, perhaps even 4096 if they are valid after a certain date.

Since SMIME keys themselves can also have value long after expiration of the certificate, there may even be a case that we should raise the key size for subscriber certificates. Open to thoughts on that topic

**Comment by timfromdigicert** [comment]
Make a plan to get off your RSA keys soon. Use all your time and effort on that, because increasing the size will not help you.

By the time RSA-2048 is too weak for a subscriber cert, all RSA will be too weak for a subscriber cert. They will all fall at roughly the same time.

Save time and energy, ignore every asymmetric crypto transition that is not to a quantum-safe algorithm. And yes, we might need signature preservation services to protect information in legacy content signed with what will be insecure algorithms. But that will be more about authenticity, because the confidentiality ship will have sailed at that point.

**Comment by dzacharo** [comment]
> Our current requirements sets the key size for CA certificates at 2048 bits for RSA. Given how long Root and Sub CA certificates may live, I would suggest we add language to raise this to a minimum of 3072, perhaps even 4096 if they are valid after a certain date.
> 
> Since SMIME keys themselves can also have value long after expiration of the certificate, there may even be a case that we should raise the key size for subscriber certificates. Open to thoughts on that topic

I support this change.

**Comment by dzacharo** [comment]
> Make a plan to get off your RSA keys soon. Use all your time and effort on that, because increasing the size will not help you.
> 
> By the time RSA-2048 is too weak for a subscriber cert, all RSA will be too weak for a subscriber cert. They will all fall at roughly the same time.
> 
> Save time and energy, ignore every asymmetric crypto transition that is not to a quantum-safe algorithm. And yes, we might need signature preservation services to protect information in legacy content signed with what will be insecure algorithms. But that will be more about authenticity, because the confidentiality ship will have sailed at that point.

Until we are close to that day, removing 2048 seems like an easy and uncontroversial change, seemingly easy to implement across the board.

**Comment by CBonnell** [comment]
Can the proponents of this change explain why this would be beneficial?

**Comment by romanf** [comment]
https://www.bsi.bund.de/SharedDocs/Downloads/EN/BSI/Publications/TechGuidelines/TG02102/BSI-TR-02102-1.html is also recommedning this move.

**Comment by tadahik** [comment]
let me raise two points.

1) NIST seems to be proposing migration to PQC by 2035, and allow use of 112 bit security by 2035.
https://nvlpubs.nist.gov/nistpubs/ir/2024/NIST.IR.8547.ipd.pdf
I am not sure if migration on SMIME can be finished by 2035, but It should be good to concern about it.

2) some people think, SMIME is "communication protocol". 
which means... current use of "SMIME" can be divided into two phase, communication phase (sender send e-mail to receiver), and archive phase (receiver store e-mail).

> Since SMIME keys themselves can also have value long after expiration of the certificate

I had discussion with some folk that, this problem happens, because of we are using same key for archiving and communication.  They have quite different characteristics (data on transfer, and data at rest).
In the other hand, it is hard to prevent user to use same key for both purpose.

my personal feeling is, it can be more straight forward and simple if we treat SMIME as communication protocol, and encourage receiver to distinguish with achieve protocol.

**Comment by romanf** [comment]
I agree with Tadahiko and to make matters worse, many S/MIME certificates also alllow for digital signature... IMHO these use-cases should be split into transport encryption, transport authenticity and long-term storage.

**Comment by srdavidson** [comment]
See draft at https://github.com/cabforum/smime/compare/be9a18ab2b48eb0cbff41d3a268202f700c06c05...a90411aa2a7d2279692c21a35597a950cc9f37c7
